### PR TITLE
Add a first test to scopes/collections ISGR

### PR DESCRIPTION
**Pull request by gilkal**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- Add a first test for ISGR, push/pull replication and multi collections
-

### CODE REVIEW

File: testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/test_scopes_and_collections.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    user1_auth = sgs["sg1"]["user"], password
+    user2_auth = sgs["sg2"]["user"], password
+
+    # 2. Upload documents to the pushing SGW and make sure that there are no document on the passive SGW
+    uploaded_for_pull = sg_client.add_docs(url=sg1_url, db=sg1["db"], number=num_of_docs, id_prefix=pull_replication_prefix, channels=["A"], auth=user1_auth, scope=scope, collection=collection)
+    with pytest.raises(Exception) as e:
+        sg_client.get_doc(sg2_url, sg2["db"], uploaded_for_pull[0]["id"], rev=uploaded_for_pull[0]["rev"], auth=user2_auth, scope=scope, collection=collection)
+    e.match("Not Found")
+
+    # 3. start a pull continous replication sg1 <- sg2
```
Comment (muntazim-aliCB) [inline testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/test_scopes_and_collections.py:152]: minor, but should this comment read "sg1 -> sg2" as sg2 is the local db pulling from sg1?

File: testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/test_scopes_and_collections.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        continuous=False,
+        collections_enabled=True
+    )
+    admin_client_2.wait_until_sgw_replication_done(sg2["db"], replicator2_id, read_flag=True, max_times=3000)
+
+    # 3. Check that the documents were replicated to sgw2
+    for i in range(0, num_of_docs - 1):
+        sg_client.get_doc(sg2_url, sg2["db"], uploaded_for_pull[i]["id"], rev=uploaded_for_pull[i]["rev"], auth=user2_auth, scope=scope, collection=collection)
+
+    # 4. Upload new documents to sgw1
```
Comment (muntazim-aliCB) [inline testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/test_scopes_and_collections.py:169]: is it worth having the test have documents in both the previous collection and the newly added collection?

File: testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/test_scopes_and_collections.py (hunk 1 above)
Comment (gilkal) [reply testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/test_scopes_and_collections.py:152]: Thanks, fixed.

File: testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/test_scopes_and_collections.py (hunk 2 above)
Comment (gilkal) [reply testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/test_scopes_and_collections.py]: Yes, it's worth, good idea :) Fixed.

**Review by muntazim-aliCB — APPROVED** [review]

**Pull request merged**